### PR TITLE
Reindent the whole role

### PR DESCRIPTION
**Pull request by mscherer**
Also fix a few bugs:
- wrong permission on documentroot
- spurious conditional for enabling mod_security

### CODE REVIEW

File: tasks/freeipa.yml [hunk 1]
```diff
@@ -1,8 +1,17 @@
...
+  command: ipa-getcert request -f {{ cert_file }} -k {{ key_file }} -D {{ ansible_fqdn }} -K host/{{ ansible_fqdn }}
+  args:
+    creates: "{{ key_file }}"
 
-- file: state=link src=/etc/ipa/ca.crt dest=/etc/pki/tls/certs/{{ _website_domain }}.ca.crt
+- name: Link freeipa CA to httpd directory
+  file:
+    state: link
+    src: /etc/ipa/ca.crt
+    dest: /etc/pki/tls/certs/{{ _website_domain }}.ca.crt
```
Comment (duck-rh) [inline tasks/freeipa.yml]: I think it's better to add quotes whenever you use variables

Comment (duck-rh) [reply tasks/freeipa.yml]: s/better/safest/

File: tasks/main.yml [hunk 2]
```diff
@@ -55,8 +71,15 @@
...
 - include: mod_speling.yml
 
-- set_fact: port=80
-- template: src=vhost.conf dest=/etc/httpd/conf.d/{{ _website_domain }}.conf owner=root group=apache mode=0644
+- set_fact:
+    port: 80
+
+- template:
+    src: vhost.conf
+    dest: /etc/httpd/conf.d/{{ _website_domain }}.conf
```
Comment (duck-rh) [inline tasks/main.yml]: same

File: tasks/main.yml [hunk 3]
```diff
@@ -84,16 +107,22 @@
...
   when: use_hidden_service
 
-- set_fact: port=443
-- template: src=vhost.conf dest=/etc/httpd/conf.d/{{ _website_domain }}_ssl.conf owner=root group=apache mode=0644
+- set_fact:
+    port: 443
+
+- template:
+    src: vhost.conf
+    dest: /etc/httpd/conf.d/{{ _website_domain }}_ssl.conf
```
Comment (duck-rh) [inline tasks/main.yml]: same

File: tasks/main.yml [hunk 4]
```diff
@@ -84,16 +107,22 @@
...
 # due to https://github.com/ansible/ansible/issues/14472
-- set_fact: _force_tls=False
-  name: Clean force_tls
-
-- set_fact: _use_tls=False
-  name: Clean use_tls
+- set_fact:
+    _force_tls: False
+    _use_tls: False
+  name: Clean force_tls and use_tls
```
Comment (duck-rh) [inline tasks/main.yml:128]: putting the name before would improve readability

File: tasks/mod_security.yml [hunk 5]
```diff
@@ -1,12 +1,16 @@
...
   notify: verify config and restart httpd
   name: Install mod_security packages
 
-- lineinfile: dest=/etc/httpd/conf.d/mod_security.conf regexp='^(\s*)SecRuleEngine ' line='\1SecRuleEngine {{ 'DetectionOnly' if mod_sec_detection_only else 'On' }}'
+- lineinfile:
+    dest: /etc/httpd/conf.d/mod_security.conf
+    regexp: "^(\\s*)SecRuleEngine "
+    line: "\\1SecRuleEngine {{ 'DetectionOnly' if mod_sec_detection_only else 'On' }}"
   notify: verify config and restart httpd
-  when: mod_sec_detection_only is True
```
Comment (duck-rh) [inline tasks/mod_security.yml:11]: side-effect change I suppose

File: tasks/password_present.yml [hunk 6]
```diff
@@ -1,6 +1,15 @@
 - name: Create the htpasswd file
-  htpasswd: path=/etc/httpd/{{ _website_domain }}.htpasswd name={{ website_user }} password={{ website_password }} owner=root group=apache mode=0640
+  htpasswd:
+    path: /etc/httpd/{{ _website_domain }}.htpasswd
```
Comment (duck-rh) [inline tasks/password_present.yml]: same

File: tasks/password_present.yml [hunk 7]
```diff
@@ -1,6 +1,15 @@
...
+    group: apache
+    mode: 0640
 
 - name: Deploy the website password protection
-  template: src=password.conf dest=/etc/httpd/conf.d/{{ _website_domain }}.conf.d/password.conf
+  template:
+    src: password.conf
+    dest: "/etc/httpd/conf.d/{{ _website_domain }}.conf.d/password.conf"
   notify: verify config and restart httpd
+  no_log: True
```
Comment (duck-rh) [inline tasks/password_present.yml:15]: good catch but outside this PR

File: tasks/redirect.yml [hunk 8]
```diff
@@ -1,3 +1,5 @@
-- template: src=redirect.conf dest=/etc/httpd/conf.d/{{ _website_domain }}.conf.d/redirect.conf
+- template:
+    src: redirect.conf
+    dest: /etc/httpd/conf.d/{{ _website_domain }}.conf.d/redirect.conf
```
Comment (duck-rh) [inline tasks/redirect.yml]: same

File: tasks/redirect_to_https.yml [hunk 9]
```diff
@@ -1,3 +1,5 @@
-- template: src=redirect_to_https.conf dest=/etc/httpd/conf.d/{{ _website_domain }}.conf.d/redirect_to_https.conf
+- template:
+    src: redirect_to_https.conf
+    dest: /etc/httpd/conf.d/{{ _website_domain }}.conf.d/redirect_to_https.conf
```
Comment (duck-rh) [inline tasks/redirect_to_https.yml]: same

File: tasks/redirects.yml [hunk 10]
```diff
@@ -1,3 +1,5 @@
-- template: src=redirects.conf dest=/etc/httpd/conf.d/{{ _website_domain }}.conf.d/redirects.conf
+- template:
+    src: redirects.conf
+    dest: /etc/httpd/conf.d/{{ _website_domain }}.conf.d/redirects.conf
```
Comment (duck-rh) [inline tasks/redirects.yml]: same

File: tasks/reverse_proxy.yml [hunk 11]
```diff
@@ -1,2 +1,4 @@
-- template: src=reverse_proxy.conf dest=/etc/httpd/conf.d/{{ _website_domain }}.conf.d/reverse_proxy.conf
+- template:
+    src: reverse_proxy.conf
+    dest: /etc/httpd/conf.d/{{ _website_domain }}.conf.d/reverse_proxy.conf
```
Comment (duck-rh) [inline tasks/reverse_proxy.yml]: same

File: tasks/server_alias.yml [hunk 12]
```diff
@@ -1,3 +1,5 @@
 - name: Install server alias configuration
-  template: src=server_alias.conf dest=/etc/httpd/conf.d/{{ _website_domain }}.conf.d/server_alias.conf
+  template:
+    src: server_alias.conf
+    dest: /etc/httpd/conf.d/{{ _website_domain }}.conf.d/server_alias.conf
```
Comment (duck-rh) [inline tasks/server_alias.yml]: same

**Review by duck-rh — CHANGES_REQUESTED** [review]
well, quite a boring rework I suppose, but really makes things much more readable

File: tasks/mod_security.yml (hunk 5 above)
Comment (mscherer) [reply tasks/mod_security.yml:11]: that's written in the PR: "spurious conditional for enabling mod_security".

Comment (duck-rh) [reply tasks/mod_security.yml:11]: true, missed reading the description. nevertheless it should be in a separate PR, or in a patchset PR so you can review each part individually

**Pull request merged**